data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -61,16 +62,42 @@ export const setRevisionMetadata = async (
 ) => {
     const curatorEmail = request.body.curator.email;
 
-    // Find the case if it already exists so we can update its existing
+    // Find the case(s) if it/they already exists so we can update existing
     // metadata.
-    const c = await getCase(request);
+    if (request.body.cases) {
+        // Batch upsert.
+        const existingCases = await findCasesWithCaseReferenceData(request);
```

Comment:
Small question (don't worry about it if it's a PITA to measure) -- did we confirm that the MongoDB call was what made this slow? Seems very likely, but I wonder whether these in-memory operations took much time in comparison for the large (50k) invocations.

Reply:
nope -- I didn't do full profiling of where the bottleneck was, but from monitoring the server logs, it looks like a *lot* of the slowdown was MongoDB. It seemed like its page size was ~100 docs, and it was paging slowly. Not sure if there's some optimization we can do there?

Reply:
SG. I bet we'd see a speedup if we set `ordered` false in the `bulkWrite`. For some reason I'd internalized that it was false by default, but that's not the case. I can check that out separately.

Reply:
good idea!